### ISSUE TITLE
makeLinkEnabler does not allow marking specific links as "normal" links

### ISSUE THREAD

**Issue by tijmenbaarda** [CentreForDigitalHumanities/backbone-util#4]
In the [the EDPOP vre](https://github.com/UUDigitalHumanitieslab/EDPOP), we have certain internal links in the UI that are not handled by the SPA but instead by the backend. There are also a number of `<a>` elements that act like buttons (they have `#` as their anchor and they include the `role="button"` property). I don't see a way in `makeLinkEnabler` to exclude specific links. I would propose the following:

* Exclude `<a>` elements with `class="backbone-util-no-spa"` (better suggestions welcome)
* Exclude `<a>` elements with `role="button"`, because they should never act as links

@jgonggrijp : if you agree, I will create a pull request for these changes.

**Comment by jgonggrijp** [comment]
Regarding links that should be handled by the backend rather than the SPA, I think it will suffice if you include the domain and a scheme in the URL. For example, change `<a href="/login">` into `<a href="//localhost:8000/login">`.

As for an `<a role=button href="#">`, I agree that it should not be intercepted by the link enabler. But what if the `href` is something more elaborate? Is there really such a strict semantic that anything with `role=button` should never lead to route changes?

**Comment by tijmenbaarda** [comment]
> As for an `<a role=button href="#">`, I agree that it should not be intercepted by the link enabler. But what if the `href` is something more elaborate? Is there really such a strict semantic that anything with `role=button` should never lead to route changes?

Now I think of it more carefully: I think you are right. Take GrETEL, where the application state can be very carefully reproduced from the URL. This application also tracks the configuration, which is set using buttons, in the URL. These buttons are clearly not links, but they do change the URL.

Would you rather make an exception for `<a href="#">`?

**Comment by jgonggrijp** [comment]
Yes, I think an exception for `href="#"` (also with `<area>` elements) would be useful.
